File: simulation/g4simulation/TestEvtGen/HepMC3/test/testIO2.cc
// -*- C++ -*-
//
// This file is part of HepMC
// Copyright (C) 2014-2019 The HepMC collaboration (see AUTHORS for details)
//
#include "HepMC3/GenEvent.h"
#include "HepMC3/ReaderRootTree.h"
#include "HepMC3/WriterRootTree.h"
#include "HepMC3/ReaderAsciiHepMC2.h"
#include "HepMC3/WriterAsciiHepMC2.h"
#include "HepMC3TestUtils.h"
using namespace HepMC3;
int main()
{
    ReaderAsciiHepMC2 inputA("inputIO2.hepmc");
    if(inputA.failed()) return 1;
    WriterRootTree       outputA("frominputIO2.root");
    if(outputA.failed())  return 2;
    while( !inputA.failed() )
    {
        GenEvent evt(Units::GEV,Units::MM);
        inputA.read_event(evt);
        if( inputA.failed() )  {
            printf("End of file reached. Exit.\n");
            break;
        }
        outputA.write_event(evt);
        evt.clear();
    }
    inputA.close();
    outputA.close();

    ReaderRootTree inputB("frominputIO2.root");
    if(inputB.failed()) return 3;
    WriterAsciiHepMC2       outputB("fromfrominputIO2.hepmc");
    if(outputB.failed()) return 4;
    while( !inputB.failed() )
    {
        GenEvent evt(Units::GEV,Units::MM);
        inputB.read_event(evt);
        if( inputB.failed() )  {
            printf("End of file reached. Exit.\n");
            break;
        }
        outputB.write_event(evt);
        evt.clear();
    }
    inputB.close();
    outputB.close();
    return COMPARE_ASCII_FILES("fromfrominputIO2.hepmc","inputIO2.hepmc");
}

File: simulation/g4simulation/TestEvtGen/HepMC3/python/test/test_IO2.py
from pyHepMC3TestUtils import update_path, python_label
import sys

sys.path = update_path()


from pyHepMC3TestUtils import COMPARE_ASCII_FILES
from pyHepMC3 import HepMC3 as hm
from pyHepMC3.rootIO import HepMC3 as hmrootIO

print(dir(hmrootIO))


def test_IO2():
    inputA = hm.ReaderAsciiHepMC2("inputIO2.hepmc")
    if inputA.failed():
        sys.exit(1)
    outputA = hmrootIO.WriterRootTree(python_label() + "frominputIO2.root")
    if outputA.failed():
        sys.exit(2)
    while not inputA.failed():
        evt = hm.GenEvent()
        inputA.read_event(evt)
        if inputA.failed():
            print("End of file reached. Exit.\n")
            break
        outputA.write_event(evt)
        evt.clear()
    inputA.close()
    outputA.close()

    inputB = hmrootIO.ReaderRootTree(python_label() + "frominputIO2.root")
    if inputB.failed():
        sys.exit(3)
    outputB = hm.WriterAsciiHepMC2(python_label() + "fromfrominputIO2.hepmc")
    if outputB.failed():
        sys.exit(4)
    while not inputB.failed():
        evt = hm.GenEvent()
        inputB.read_event(evt)
        if inputB.failed():
            print("End of file reached. Exit.\n")
            break
        outputB.write_event(evt)
        evt.clear()
    inputB.close()
    outputB.close()
    assert 0 == COMPARE_ASCII_FILES(python_label() + "fromfrominputIO2.hepmc", "inputIO2.hepmc")
    return 0


if __name__ == "__main__":
    result = 1
    try:
        result = test_IO2()
    except:
        result = 1
    sys.exit(result)
